src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -1078,14 +1079,19 @@ private Task ReadFromRowSourceAsync(CancellationToken cts)
             if (_isAsyncBulkCopy && _dbDataReaderRowSource != null)
             {
                 // This will call ReadAsync for DbDataReader (for SqlDataReader it will be truly async read; for non-SqlDataReader it may block.)
-                return _dbDataReaderRowSource.ReadAsync(cts).ContinueWith((t) =>
-                {
-                    if (t.Status == TaskStatus.RanToCompletion)
+                return _dbDataReaderRowSource.ReadAsync(cts).ContinueWith(
+                    static (Task<bool> task, object state) =>
                     {
-                        _hasMoreRowToCopy = t.Result;
-                    }
-                    return t;
-                }, TaskScheduler.Default).Unwrap();
+                        SqlBulkCopy sqlBulkCopy = (SqlBulkCopy)state;
```

Comment:
why are we creating a new instance of SqlBulkCopy? _hasMoreRowToCopy is the outer variable in this case. Are we trying to prevent  getting a wrong value of _hasMoreRowTocpy changed by other threads?

Reply:
`_hasMoreRowToCopy ` is a field and is implicitly `this._hasMoreRowToCopy` when used in the original. When changed to static there is no `this` reference and the sqlBulkCopy variable is the same object passed in as state.